src/Nethermind/Nethermind.Core/Specs/ForkActivation.cs

```diff
@@ -43,20 +50,9 @@ public override string ToString()
 
     public int CompareTo(ForkActivation other)
     {
-        if (BlockNumber < other.BlockNumber)
-            return -1;
-        if (BlockNumber > other.BlockNumber)
-            return 1;
-        if (BlockNumber == other.BlockNumber)
-        {
-            if (Timestamp < other.Timestamp)
-                return -1;
-            if (Timestamp > other.Timestamp)
-                return 1;
-            if (Timestamp == other.Timestamp)
-                return 0;
-        }
-        throw new InvalidOperationException("This is unreachable exception");
+        return Timestamp is null || other.Timestamp is null
+            ? BlockNumber.CompareTo(other.BlockNumber)
+            : Timestamp.Value.CompareTo(other.Timestamp.Value);
```

Comment:
from the notes from the geth team meeting. they said they will not go back to blocknumber forking again. it will always be timestamp. so we do not expect blocknumber forking to come again